client/modules/IDE/components/FileNode.jsx

```diff
@@ -77,6 +77,9 @@ export class FileNode extends React.Component {
     ) {
       this.props.updateFileName(this.props.id, this.originalFileName);
     }
+    if (this.props.name.length) {
```

Comment:
i think this needs to come first in the `if` statements. also i think the logic is messed up here—it should be `if` and `if else` statements here so the file name doesn't get updated multiple times.